[#47993] Support Project status explanation changes when rendering project activity

aaron-contreras opened the pull request:
Adds rendering ability for project status explanation changes on activity pages

Work package: https://community.openproject.org/work_packages/47993

aaron-contreras committed:
Register status_explanation as a field to be rendered

Adds status_explanation as part of the rendering of activities
for project attributes.

aaron-contreras commented on config/locales/en.yml:
```diff
@@ -551,6 +551,15 @@ en:
           false: "private"
         queries: "Queries"
+        status_code: 'Status'
+        status_explanation: 'Project status description'
```
Until the required PR is merged into `dev`, the only real diff is the line I'm commenting on.

cbliard commented on app/controllers/journals_controller.rb:
```diff
@@ -101,9 +101,8 @@ def field_param
-  # Is this a valid field for diff'ing?
-  def valid_field?
-    field_param == 'description'
+  def valid_field_for_diffing?
```
I appreciate that you removed the comment and updated the method name to make the intention clearer :+1:

cbliard commented on spec/features/activities/project_attributes_activity_spec.rb:
```diff
@@ -87,6 +87,7 @@
 
       # own fields
       expect(page).to have_selector('li', text: "Name changed from #{previous_project_attributes['name']} to #{project.name}")
+      expect(page).to have_selector('li', text: 'Project status description set (Details)')
```
Can you move this assertion under the `expect(page).to have_selector('li', text: 'Project marked as template')` so that the order is the same for assignments and assertions?

aaron-contreras replied:
I sure can!

cbliard reviewed (approved):
Thanks. I tested it and it works as expected.

I only have a nitty comment about assertions order and then it will be good to merge.
Once fixed feel free to merge it and update the work package status.

aaron-contreras committed:
Re-order assignment and assertion of status explanation

In order to keep consistency for readability's sake between the assignment
of updated fields and the assertions, making sure to keep these two pairs in
order

cbliard committed:
Merge branch 'dev' into implementation/47993-support-project-status-explanation-changes-when-rendering-project-activity
